pkg/engine/variables/vars.go

```diff
@@ -308,82 +308,149 @@ func DefaultVariableResolver(ctx context.EvalInterface, variable string) (interf
 	return ctx.Query(variable)
 }
 
-func substituteVariablesIfAny(log logr.Logger, ctx context.EvalInterface, vr VariableResolver) jsonUtils.Action {
-	isDeleteRequest := isDeleteRequest(ctx)
-	return jsonUtils.OnlyForLeafsAndKeys(func(data *jsonUtils.ActionData) (interface{}, error) {
-		value, ok := data.Element.(string)
-		if !ok {
-			return data.Element, nil
-		}
+func find_next_variable(s string) []int {
```

Comment:
Clarifying the behavior we want:

`{{- x1 }}`: shallow substitute x1 i.e. substitute once and break
`{{ {{ x1 }} {{- x2 }}  }}`: deep substitute x1; shallow substitute x2

Does that match what you had in mind?

Reply:
yes